src/control/system/membership.go

```diff
@@ -447,33 +447,37 @@ func (m *Membership) CheckHosts(hosts string, ctlPort int) (*RankSet, *hostlist.
 	return rs, missHS, nil
 }
 
+func (m *Membership) handleRankDown(evt *events.RASEvent) {
+	ei := evt.GetRankStateInfo()
+	if ei == nil {
+		m.log.Error("no extended info in RankDown event received")
+		return
+	}
+	m.log.Debugf("processing RAS event %q from rank %d on host %q",
+		evt.Msg, evt.Rank, evt.Hostname)
+
+	// TODO: sanity check that the correct member is being updated by
+	// performing lookup on provided hostname and matching returned
+	// addresses with the member address with matching rank.
+
+	if err := m.UpdateMemberStates(MemberResults{
+		NewMemberResult(Rank(evt.Rank),
+			errors.Wrap(ei.ExitErr, evt.Msg),
+			MemberStateErrored),
+	}, true); err != nil {
+		m.log.Errorf("updating member states: %s", err)
+		return
+	}
+
+	member, _ := m.Get(Rank(evt.Rank))
```

Comment:
Don't want to check the error here? Is that because we expect to return above if the rank isn't found in the membership?